File: tests/openmp_examples/devices/async_target.2.f90
!$omp    declare target (init)
!$omp    target data map(v1,v2)
!$omp    task shared(v1,v2) depend(out: N)
!$omp       target device(idev)
!$omp       end target
!$omp    end task
!$omp    task shared(v1,v2,p) depend(in: N)
!$omp       target device(idev) map(from: p)
!$omp          parallel do
!$omp       end target
!$omp    end task
!$omp    taskwait
!$omp    end target data

File: tests/openmp_examples/devices/async_target.2.c
#pragma omp begin declare variant match(device={kind(host)})
#pragma omp end declare variant
#pragma omp begin declare variant match(device={kind(nohost)})
#pragma omp end declare variant
#pragma omp begin declare target
#pragma omp end declare target
#pragma omp task shared(v1, v2) depend(out: v1, v2)
#pragma omp target device(dev) map(v1, v2)
#pragma omp task shared(v1, v2, p) depend(in: v1, v2)
#pragma omp target device(dev) map(to: v1, v2) map(from: p[0:N])
#pragma omp parallel for
#pragma omp taskwait
